src/AmplienceImageStudio.ts

```diff
@@ -163,48 +227,124 @@ class AmplienceImageStudioInstance<T> {
     return promise;
   }
 
-  protected handleEvent(event: { data: ImageStudioEvent }) {
-    if (event.data?.exportImageInfo) {
-      this.handleExportedImage(event.data?.exportImageInfo);
+  protected handleEvent(event: MessageEvent) {
+    if (!(event.data instanceof Object)) {
+      return; // any messages without data being an object can be discarded
     }
-
-    if (event.data?.connect && !this.isActive) {
-      this.handleActivate();
+    // for any events that don't contain the `type` var, these conform to legacy structure.
+    let eventData: ImageStudioEvent | null;
+    if ('type' in event.data) {
```

Comment:
Yeah, i thought about this - but given the amount of work thats been involved upto now, i was reluctant to complicate it further. It needed more thought, and right now this approach is alot more flexible, extensible and easy to deprecate in future that i couldnt justify or think of a reason why we might need to track version history.

If we added versioning, i was conscious our code would end up littered with blocks like `if (messaging.version > 2){ do something here; } else { do that }` and this can be hard to track, or even contemplate trying to strip out code in future. I would prefer sdk developers to deprecate message types, and introduce new ones if this ever is required.

You will also notice that i chose explicit strings for enum values so that they serialise/deserialise neatly in future, ordering of these enums doesnt matter anymore and they translate nicely over post message irrespective of the version of the client/sdk we are using on either end.